src/trace/xray-service.ts

```diff
@@ -187,13 +196,29 @@ export class XrayService {
     if (lastPart.length !== 24) return;
 
     // We want to turn the last 63 bits into a decimal number in a string representation
-    // Unfortunately, all numbers in javascript are represented by float64 bit numbers, which
-    // means we can't parse 64 bit integers accurately.
-    const hex = new BigNumber(lastPart, 16);
-    if (hex.isNaN()) return;
-
-    // Toggle off the 64th bit
-    const last63Bits = hex.mod(new BigNumber("8000000000000000", 16));
-    return last63Bits.toString(10);
+    try {
+      return (BigInt("0x" + lastPart) % BigInt("0x8000000000000000")).toString(10);
```

Comment:
Let's add an env var to understand what is this?

Reply:
thought about it. will do now.